### DISCUSSION TITLE
Content Security Policy in Browser

### DISCUSSION THREAD

**Discussion by edward221096** [kimai/kimai#4212] [On-Premise: Installation & Updates]
### Describe the issue

Hi Kimai Support,

I would like to ask if there are any guidelines when we are configuring the Content Security Policy in Kimai. 

We are having a problem with the Javascript when we access Kimai externally. The javascript was messed up. But when we access internally in our internal network the Javascript is working fine. You can reference to the attached screenshot:

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.0.29

### How do you run Kimai?

Virtual Server or alike

### Which PHP version are you using?

8.1

### Screenshots

![Cannot Click buttons](https://github.com/kimai/kimai/assets/62000583/30f0dd2c-9c67-4921-a138-ade66449d7d3)
![Error in Inspect Element External Access](https://github.com/kimai/kimai/assets/62000583/d4181be6-3337-4780-ac8a-a673dd2c24f0)
![JS messed up](https://github.com/kimai/kimai/assets/62000583/44c2d73e-718d-4e0e-aa0b-a8202726c385)

**Comment by kevinpapst** [comment]
No, there are no guidelines. But adding `'unsafe-inline'` sounds like a good idea:
```
add_header "Content-Security-Policy" "default-src https: data: 'unsafe-eval' 'unsafe-inline';";
```
Please note: the above is not enough, you need way more like `frame-ancestors 'none'; object-src 'self'; base-uri 'none'; plugin-types 'application/pdf';`.

I'd appreciate if you could share a working version, if you are successful.

But I don't see why `internal` vs `external` should make any difference. Are you using different Virtual Hosts for those versions?
